api/src/test/java/jakarta/data/displaynamegeneration/ReplaceCamelCaseAndUnderscore.java

```diff
@@ -0,0 +1,88 @@
+package jakarta.data.displaynamegeneration;
+
+import org.junit.jupiter.api.DisplayNameGenerator;
+
+import java.lang.reflect.Method;
+
+public class ReplaceCamelCaseAndUnderscore extends DisplayNameGenerator.Standard {
+
+    public ReplaceCamelCaseAndUnderscore() {
+        super();
+    }
+
+    @Override
+    public String generateDisplayNameForClass(Class<?> testClass) {
+        return replaceCamelCaseAndUnderscore(super.generateDisplayNameForClass(testClass));
+    }
+
+    @Override
+    public String generateDisplayNameForNestedClass(Class<?> nestedClass) {
+        return replaceCamelCaseAndUnderscore(super.generateDisplayNameForNestedClass(nestedClass));
+    }
+
+    @Override
+    public String generateDisplayNameForMethod(Class<?> testClass, Method testMethod) {
+        return replaceCamelCaseAndUnderscore(testMethod.getName()) + DisplayNameGenerator.parameterTypesAsString(testMethod);
+    }
+
+    public String replaceCamelCaseAndUnderscore(String input) {
+        StringBuilder result = new StringBuilder();
+        /*
+         * Each method name starts with "should" then the displayed name starts with "Should"
+         * */
+        result.append(Character.toUpperCase(input.charAt(0)));
+
+        /*
+         * There are 2 groups of method name: with and without underscore
+         * */
+        if (input.contains("_")) {
+            boolean insideUnderscores = false;
+            for (int i = 1; i < input.length(); i++) {
```

Comment:
Couldn't this be simplified using some regular expressions?  For example 
```java
input.replaceAll("([A-Z])", " $1"); // would replace all capital letters with a prepended space
```

Reply:
It would if we decided to keep capital letters but that's not what we want to achieve. We replace capital letters (1st letter excluded, we start at 1 instead of 0) with prepended space and lower case as you can see in this example you defined here #82 .

```java
@DisplayNameGeneration(ReplaceCamelCase.class)
class LimitTest {
    @Test
    void shouldReturnErrorWhen_maxResults_IsNegative() {
      ...
    }
    @Test
    void shouldCreateLimitWithRange() {
      ...
    }
}
```

```
- Should return error when maxResults is negative
- Should create limit with range
```

Reply:
We also have to handle '_', that's why I am using a for loop while applying some rules related to this character.